esp-hal/src/soc/esp32s2/gpio.rs

```diff
@@ -314,6 +314,106 @@ pub enum OutputSignal {
     GPIO             = 256,
 }
 
+macro_rules! rtcio_analog {
+    ( @ignore $rue:literal ) => {};
+
+    (
+        $pin_num:expr, $pin_reg:expr, $prefix:pat, $hold:ident
+    ) => {
+        impl $crate::gpio::RtcPin for GpioPin<$pin_num> {
+            fn rtc_number(&self) -> u8 {
+                $pin_num
+            }
+
+            /// Set the RTC properties of the pin. If `mux` is true then then pin is
+            /// routed to RTC, when false it is routed to IO_MUX.
+            fn rtc_set_config(&mut self, input_enable: bool, mux: bool, func: $crate::gpio::RtcFunction) {
+                enable_iomux_clk_gate();
+
+                // disable input
+                paste::paste!{
+                    unsafe { $crate::peripherals::RTC_IO::steal() }
+                        .$pin_reg.modify(|_,w| unsafe {
+                            w.[<$prefix fun_ie>]().bit(input_enable);
+                            w.[<$prefix mux_sel>]().bit(mux);
+                            w.[<$prefix fun_sel>]().bits(func as u8)
+                        });
+                }
+            }
+
+            fn rtcio_pad_hold(&mut self, enable: bool) {
+                unsafe { $crate::peripherals::LPWR::steal() }
+                    .pad_hold()
+                    .modify(|_, w| w.$hold().bit(enable));
+            }
+        }
+
+        impl $crate::gpio::RtcPinWithResistors for GpioPin<$pin_num>
+        {
+            fn rtcio_pullup(&mut self, enable: bool) {
+                let rtcio = unsafe { $crate::peripherals::RTC_IO::steal() };
+
+                paste::paste! {
+                    rtcio.$pin_reg.modify(|_, w| w.[< $prefix rue >]().bit(enable));
+                }
+            }
+
+            fn rtcio_pulldown(&mut self, enable: bool) {
+                let rtcio = unsafe { $crate::peripherals::RTC_IO::steal() };
+
+                paste::paste! {
+                    rtcio.$pin_reg.modify(|_, w| w.[< $prefix rde >]().bit(enable));
+                }
+            }
```

Comment:
this looks different for (apparently) no reason for S3

Reply:
oops